### ISSUE TITLE
Add unit tests

### ISSUE THREAD

**Issue by gianantoniopini** [manzinello/mailgo#107]
It would be good to start adding some unit tests to this useful library.

I would be willing to work on this and submit a PR. For example I could start writing some unit tests for the mailgoDirectRender and mailgoValidateEmail functions, using Typescript and the React Testing Library.

Would this be a welcome addition?

**Comment by manzinello** [comment]
> I would be willing to work on this and submit a PR. For example I could start writing some unit tests for the mailgoDirectRender and mailgoValidateEmail functions, using Typescript and the React Testing Library.

I think that it is a very important point, thank you!

> Would this be a welcome addition?

Absolutely yes! Proceed with the PR when you want, thank you so much!

**Comment by gianantoniopini** [comment]
Ok, I will start looking into this. Thanks.

**Comment by manzinello** [comment]
Hi @gianantoniopini, it seems that there is something that isn't working properly with tests (window and document). I've just updated the dependencies. I'm sorry and I can't work on this at the moment, can you help me? https://github.com/manzinello/mailgo/actions/runs/895410024 Otherwise I will go in depth with this problem as soon as I can. Thank you so much

**Comment by gianantoniopini** [comment]
@manzinello No problem, I will check this.

**Comment by manzinello** [comment]
Thank you so much!

**Comment by gianantoniopini** [comment]
The issue was related to the update of Jest to version 27. In this new version of Jest the default test environment has been changed from "jsdom" to "node". See this https://jestjs.io/blog/2021/05/25/jest-27#flipping-defaults .
We were affected by this change because we use the DOM APIs and did not have the test environment explicitly configured. So I have configured "testEnvironment": "jsdom" in the jest.config.js file.

I will create a PR for this change.
